src/storage/mvcc/txn.rs

```diff
@@ -875,9 +910,19 @@ impl<S: Snapshot> MvccTxn<S> {
     fn check_txn_status_missing_lock(
         &mut self,
         primary_key: Key,
+        mismatch_lock: Option<Lock>,
         action: MissingLockAction,
     ) -> Result<TxnStatus> {
         MVCC_CHECK_TXN_STATUS_COUNTER_VEC.get_commit_info.inc();
+
+        if let Some(l) = mismatch_lock {
```

Comment:
I think only when there isn't a commit record and an overlapped write record, we have to do it.

Reply:
Oh, sorry.

Reply:
I changed it but I found it's the only case that `mismatch_lock` is possible to be `Some`, so no more test is added.